namesys/publisher.go

```diff
@@ -41,15 +41,9 @@ func NewRoutingPublisher(route routing.IpfsRouting) Publisher {
 
 // Publish implements Publisher. Accepts a keypair and a value,
 // and publishes it out to the routing system
-func (p *ipnsPublisher) Publish(ctx context.Context, k ci.PrivKey, value u.Key) error {
+func (p *ipnsPublisher) Publish(ctx context.Context, k ci.PrivKey, value path.Path) error {
 	log.Debugf("namesys: Publish %s", value)
 
-	// validate `value` is a ref (multihash)
-	_, err := mh.FromB58String(value.Pretty())
-	if err != nil {
-		return fmt.Errorf("publish value must be str multihash. %v", err)
-	}
-
```

Comment:
though the constructors check correctness, we can still cast `p := path.Path(str)`. So may be good to have an 

```go
func (p *path.Path) IsValid() {
	... check p is correct
}
```
and check here?